### PR TITLE
Add alternative date formatting, Add lines for insulin delivery expiration, countdown to expiration, etc

### PR DESCRIPTION
**Pull request by IsThisPaul**
Add alternative date formatting, Add lines for insulin delivery expiration, countdown to expiration, etc

Add alternative date formatting &
Add lines for insulin delivery expiration, countdown to expiration, etc
In Omnipod Dash settings view

### CODE REVIEW

File: OmniBLE/OmnipodCommon/Pod.swift [hunk 1]
```diff
@@ -79,7 +79,7 @@ public struct Pod {
     // Checked to verify it agrees with value returned by pod during the pairing process.
     public static let cannulaInsertionUnits = 0.5
 
-    public static let cannulaInsertionUnitsExtra = 0.0 // edit to add a fixed additional amount of insulin during cannula insertion
+    public static let cannulaInsertionUnitsExtra = 0.5 // edit to add a fixed additional amount of insulin during cannula insertion
```
Comment (damonbayer) [inline OmniBLE/OmnipodCommon/Pod.swift]: Unintentional?

Comment (IsThisPaul) [reply OmniBLE/OmnipodCommon/Pod.swift]: Unintentional... do I need to correct and re-commit or can you reject just that?

Comment (IsThisPaul) [reply OmniBLE/OmnipodCommon/Pod.swift]: Fixed

**Comment by ps2** [comment]
Please describe the problem you are attempting to solve with these changes. I know some would prefer to see all these fields all the time, but we’ve tried to be nice to the user and highlight those that are most useful in a given context.  Every additional piece of info on the screen makes the other info less prominent and takes more time to find what you’re looking for. 

So I would like to hear the specific use case you think isn’t currently solved.

**Comment by IsThisPaul** [comment]
@ps2, I see two primary problems being solved:

1 - When the patient leaves the house, goes on a trip, or otherwise needs to plan his/her day around pod expiration, the current display requires the patient to calculate around a fully displayed date ("Is the 23rd Wednesday or Thursday?) rather than a friendly "Tomorrow at 8:34 PM" or "Friday at 4:16 PM". Since pod expiration is never farther than 3 days away, a friendly "day of week" display is one less calculation the patient has to do to treat themselves

2 - Pod users, as we know, need to plan to have supplies on hand for pod expiration. Many users make these plans around the hard insulin delivery stop that happens at 80 hours rather than the defined "expiration" of 72 hours. This PR adds a simple to read, friendly date formatted, day and time of the insulin delivery stoppage. It also takes into account the 72 hour expiration by showing the hard stop as a countdown timer (highlighted in red) once the 72 hour expiration has passed. This allows those of us who use the 80 hour clock in our planning to have a friendly date and time displayed around which we can plan our pod changes.

**Comment by ps2** [comment]
Thanks for the submission, and describing the problems you are attempting to solve; that helps a lot with evaluating the PR. I'm not able to land this as is, for the reasons below:

> than a friendly "Tomorrow at 8:34 PM" or "Friday at 4:16 PM". Since pod expiration is never farther than 3 days away, a friendly "day of week" display is one less calculation

I agree that relative time is generally more user friendly for date representations near the current time., and that's why we already *are* using a relative date formatter, that shows "yesterday/today/tomorrow", and is locale aware. 

Your changes will add day of week (DOW) display for days > 2 days in the future or past, but it uses a fixed string that will only work for english. The addition of DOW formatting for dates outside "today/tomorrow/yesterday" might be a change that could be taken, but it should 1) not use fixed strings that are not localizable, and 2) ideally be wrapped into its own formatter, to keep the logic of detecting relative/vs fixed formatting outside of the view code.

> Pod users, as we know, need to plan to have supplies on hand for pod expiration. Many users make these plans around the hard insulin delivery stop that happens at 80 hours rather than the defined "expiration" of 72 hours...

The de-prioritization of hard-stop of insulin is intentional. In the original Omnipod design/pdm, this date is not actually shown at all. In Loop, we *do* show the hard stop time (see `serviceTimeRemainingString`) when pods are nearing expiration as we know people do often use pods all the way to 80 hours. But we also agree with the design of the Omnipod system that there is risk around this behavior (not having supplies when it runs out, and being left without a delivery system), and want to encourage the use of the 72 hours as the primary communicated "expiration" time.  It's the same reason gas gauges don't put the "empty" mark at the bottom of the measurement range, but leave a little wiggle room, and also the same reason the pod still has insulin after you reach "0" units.

**Comment by IsThisPaul** [comment]
Thanks, @PS2

> I agree that relative time is generally more user friendly for date representations near the current time., and that's why we already _are_ using a relative date formatter, that shows "yesterday/today/tomorrow", and is locale aware.
> 
> Your changes will add day of week (DOW) display for days > 2 days in the future or past, but it uses a fixed string that will only work for english. The addition of DOW formatting for dates outside "today/tomorrow/yesterday" might be a change that could be taken, but it should 1) not use fixed strings that are not localizable, and 2) ideally be wrapped into its own formatter, to keep the logic of detecting relative/vs fixed formatting outside of the view code.
> 
I'll take this into account when making future updates and be sure formatters can be localized, etc. Pro tip, thanks.

> > Pod users, as we know, need to plan to have supplies on hand for pod expiration. Many users make these plans around the hard insulin delivery stop that happens at 80 hours rather than the defined "expiration" of 72 hours...
> 
> The de-prioritization of hard-stop of insulin is intentional. In the original Omnipod design/pdm, this date is not actually shown at all. In Loop, we _do_ show the hard stop time (see `serviceTimeRemainingString`) when pods are nearing expiration as we know people do often use pods all the way to 80 hours. But we also agree with the design of the Omnipod system that there is risk around this behavior (not having supplies when it runs out, and being left without a delivery system), and want to encourage the use of the 72 hours as the primary communicated "expiration" time. It's the same reason gas gauges don't put the "empty" mark at the bottom of the measurement range, but leave a little wiggle room, and also the same reason the pod still has insulin after you reach "0" units.

This will still be my personal preference but understand why such a change shouldn't be made for general distribution. I'll close this PR and consider making these changes available as an optional patch for patients who are interested.

**Pull request closed without merging**